Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkEditor.cs

```diff
@@ -1,35 +1,66 @@
-﻿#if !COMPILER_UDONSHARP && UNITY_EDITOR && UDONSHARP
+﻿#if !COMPILER_UDONSHARP && UNITY_EDITOR
+using System.Reflection;
+
+#if UDONSHARP
 using UdonSharp;
 using UdonSharpEditor;
+#endif
+
 using UnityEditor;
 using UnityEngine;
-using VRCAudioLink;
 
-[CustomEditor(typeof(AudioLink))]
-public class AudioLinkEditor : Editor
+namespace VRCAudioLink.Editor
 {
-    public override void OnInspectorGUI()
-    {
-        if (UdonSharpGUI.DrawDefaultUdonSharpBehaviourHeader(target)) return;
-        EditorGUILayout.Space();
-        if (GUILayout.Button(new GUIContent("Link all sound reactive objects to this AudioLink", "Links all UdonBehaviours with 'audioLink' parameter to this object."))) { LinkAll(); }
-        EditorGUILayout.Space();
-        base.OnInspectorGUI();
-    }
-
-    void LinkAll()
+    [CustomEditor(typeof(AudioLink))]
+    public class AudioLinkEditor : UnityEditor.Editor
     {
-        UdonSharpBehaviour[] allUdonSharpBehaviours = FindObjectsOfType<UdonSharpBehaviour>();
-        foreach (var behaviour in allUdonSharpBehaviours)
+        public override void OnInspectorGUI()
         {
-            if (behaviour.GetType().GetField("audioLink") != null)
+#if UDONSHARP
+            if (UdonSharpGUI.DrawDefaultUdonSharpBehaviourHeader(target)) return;
+#endif
+            // don't use Camera.main here because the main camera might not have the main camera tag?
+            if (FindObjectsOfType<Camera>().Length <= 1)
```

Comment:
this is kinda wack to have in a hot-loop since the perf gets arbitrarily bad, but I guess it's not so bad considering it will only happen when the inspector is shown

Reply:
I tested this and couldn't see a slow down even with moderately sized scenes, but you are right, it's not ideal. we can probably make some assumptions and just use Camera.main